### PR TITLE
Updated main.tf.j2 as per new terraform updates.

### PR DESCRIPTION
**Pull request by anxious-bing**
Updated the file with the required_providers block. The version meta-argument needs to be included here post terraform version 0.13. Will also be updating the file provider.tf.j2.

### CODE REVIEW

File: templates/azure/main.tf.j2 [hunk 1]
```diff
@@ -1,5 +1,11 @@
 terraform {
   required_version = ">= 0.12"
+  required_providers {
+    az = {
```
Comment (sshnaidm) [inline templates/azure/main.tf.j2]: Please change it to `azurerm`:

https://www.terraform.io/docs/language/providers/requirements.html#local-names

> Users of a provider can choose any local name for it. However, nearly every provider has a preferred local name, which it uses as a prefix for all of its resource types. (For example, resources from hashicorp/aws all begin with aws, like aws_instance or aws_security_group.)
> 
> Whenever possible, you should use a provider's preferred local name. This makes your configurations easier to understand, and lets you omit the provider meta-argument from most of your resources. (If a resource doesn't specify which provider configuration to use, Terraform interprets the first word of the resource type as a local provider name.)

**Review by sshnaidm — CHANGES_REQUESTED** [review]

**Comment by sshnaidm** [comment]
Please fix also errors from failing Azure job in CI.

**Comment by anxious-bing** [comment]
> Please fix also errors from failing Azure job in CI.

Done.

File: templates/azure/main.tf.j2 [hunk 2]
```diff
@@ -1,5 +1,11 @@
 terraform {
   required_version = ">= 0.12"
```
Comment (sshnaidm) [inline templates/azure/main.tf.j2]: This config is supported form 0.13, so let's bump required version to >=0.13

**Review by sshnaidm — CHANGES_REQUESTED** [review]

**Review by sshnaidm — APPROVED** [review]

**Pull request merged**